float64 start_time
float64 finish_time
